File: backend/src/main/java/com/stepx/stepx/dto/ShoeDTO.java
package com.stepx.stepx.dto;

import java.math.BigDecimal;
import java.util.List;

public record ShoeDTO(
    Long id,
    String name,
    String shortDescription,
    String longDescription,
    BigDecimal price,
    String brand,
    String category,
    String imageUrl1,
    String imageUrl2,
    String imageUrl3,
    List<ShoeSizeStockDTO> sizeStocks,
    List<ReviewDTO> reviews
) {}

File: frontend/src/app/dtos/shoe.dto.ts
import { ReviewDTO } from './review.dto'; 
import { ShoeSizeStockDTO } from './shoesizestock.dto'; 


export interface ShoeDTO {
    id: number;                    // Long in Java mapped to number in TypeScript
    name: string;
    shortdescription: string;       // Short description of the shoe
    longDescription: string;       // Long description of the shoe
    price: number;                  // BigDecimal in Java mapped to number in TypeScript
    brand: string;
    category: string;
    imageUrl1: string;             // Image URL for the first image
    imageUrl2: string;             // Image URL for the second image
    imageUrl3: string;             // Image URL for the third image
    sizeStocks: ShoeSizeStockDTO[]; // List of available sizes and stock
    reviews: ReviewDTO[];           // List of reviews for the shoe
  }
  